lightningd/peer_control.c

```diff
@@ -189,7 +189,7 @@ void peer_fail_permanent(struct peer *peer, const u8 *msg TAKES)
 		    (int)tal_len(msg), (char *)msg);
 
 	/* We can have multiple errors, eg. onchaind failures. */
-	if (!peer->error)
+	if (peer_has_error(peer))
```

Comment:
You have omitted the not operator here !

Reply:
nice catch!